# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gettext/hotfix.patch
# Copyright (C) 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Gettext is failing on this: In function 'open',
  inlined from 'msgdomain_list_print' at write-catalog.c:223:
/usr/include/bits/fcntl2.h:51: error: call to '__open_missing_mode' declared with attribute error: open with O_CREAT in second argument needs 3 arguments

It seems that the O_CREAT flag needs additional parameters to be legal.

  Jan

diff -ur gettext-0.17/gettext-tools/src/write-catalog.c gettext-0.17-fix/gettext-tools/src/write-catalog.c
--- gettext-0.17/gettext-tools/src/write-catalog.c	2007-10-07 15:37:39.000000000 -0400
+++ gettext-0.17-fix/gettext-tools/src/write-catalog.c	2010-02-01 18:18:13.000000000 -0500
@@ -220,7 +220,7 @@
      /* Open the output file.  */
      if (!to_stdout)
	{
-	  fd = open (filename, O_WRONLY | O_CREAT);
+	  fd = open (filename, O_WRONLY | O_CREAT, S_IRUSR|S_IWUSR);
	  if (fd < 0)
	    {
	      const char *errno_description = strerror (errno);
